man/rmd/linear_reg_gls.Rmd

```diff
@@ -0,0 +1,121 @@
+```{r, child = "setup.Rmd", include = FALSE}
+```
+
+`r descr_models("linear_reg", "gls")`
+
+## Tuning Parameters
+
+This model has no tuning parameters.
+
+## Translation from parsnip to the original package
+
+```{r gls-csl}
+linear_reg() %>% 
+  set_engine("gls") %>% 
+  set_mode("regression") %>% 
+  translate()
+```
+
+
+## Preprocessing requirements
+
+There are no specific preprocessing needs. However, it is helpful to keep the clustering/subject identifier column as factor or character (instead of making them into dummy variables). See the examples in the next section. 
+
+## Other details
+
+The model can accept case weights. 
+
+With `parsnip`, we suggest using the _fixed effects_ formula method when fitting but the details of the correlation structure should be passed to `set_engine()` since it is an irregular (but required) argument:
+
+```{r}
+library(tidymodels)
+# load nlme to be able to use the `cor*()` functions
+library(nlme)
+
+data("riesby")
+
+linear_reg() %>% 
+  set_engine("gls", correlation =  corCompSymm(form = ~ 1 | subject)) %>% 
+  fit(depr_score ~ week, data = riesby)
+```
+
+When using the general tidymodels infrastructure, it may be better to use a workflow. In this case, you can add the appropriate columns using `add_variables()` then supply the typical formula when adding the model: 
+
+```r
+library(tidymodels)
+
+gls_spec <- 
+  linear_reg() %>% 
+  set_engine("gls", correlation =  corCompSymm(form = ~ 1 | subject))
+
+gls_wflow <- 
+  workflow() %>% 
+  # The data are included as-is using:
+  add_variables(outcomes = depr_score, predictors = c(week, subject)) %>% 
+  add_model(gls_spec, formula = depr_score ~ week)
+
+fit(gls_wflow, data = riesby)
+```
+
+# Degrees of freedom
+
+Note that [nlme::lme()] and [nlme::gls()] can fit the same model but will count degrees of freedom differently. If there are `n` data points, `p` fixed effects parameters, and `q` random effect parameters, the residual degrees of freedom are:
+
+* `lme`: n - p - q
+* `gls`: n - p
+
+As a result, p-values will be different. For example, we can fit the same model using different estimation methods (assuming a positive covariance value): 
+
+```{r}
+gls_fit <- 
+  linear_reg() %>% 
+  set_engine("gls", correlation =  corCompSymm(form = ~ 1 | subject)) %>% 
+  fit(depr_score ~ week, data = riesby)
+
+lme_fit <-
+  linear_reg() %>% 
+  set_engine("lme", random =  ~ 1|subject) %>% 
+  fit(depr_score ~ week, data = riesby)
+```
+
+The estimated within-subject correlations are the same:
+
+```{r}
+library(ape)
+
+# lme, use ape package:
+lme_within_sub <- varcomp(lme_fit$fit)/sum(varcomp(lme_fit$fit))
```

Comment:
a use case for `extract_fit_engine()`

Reply:
I have a separate vignette that goes over that. This is a little bit easier since it references the fit twice.